GVFS/GVFS.Common/InstallerPreRunChecker.cs

```diff
@@ -0,0 +1,212 @@
+using GVFS.Common;
+using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+
+namespace GVFS.Upgrader
+{
+    public class InstallerPreRunChecker
+    {
+        private static readonly HashSet<string> BlockingProcessSet = new HashSet<string> { "GVFS", "GVFS.Mount", "git", "ssh-agent", "bash", "wish", "git-bash" };
+
+        private ITracer tracer;
+
+        public InstallerPreRunChecker(ITracer tracer, string commandToRerun)
+        {
+            this.tracer = tracer;
+            this.CommandToRerun = commandToRerun;
+        }
+
+        protected string CommandToRerun { private get; set; }

```

Comment:
@alameenshah it doesn't really make sense to have a private getter and a protected setter. I think we need to further clean up the unit test pattern here, so please put this on your list.

Reply:
Added to the cleanup tasks.